### ISSUE TITLE
null "old" object in MemberUpdateEvent

### ISSUE THREAD

**Issue by j0rdanit0** [Discord4J/Discord4J#806]
**To Reproduce:** 
1. Start up a GatewayDiscordClient with these intents:
```java
.setEnabledIntents( IntentSet.of(
        Intent.GUILD_MEMBERS,
        Intent.GUILDS,
        Intent.GUILD_MESSAGES,
        Intent.GUILD_MESSAGE_REACTIONS,
        Intent.DIRECT_MESSAGES,
        Intent.DIRECT_MESSAGE_REACTIONS
) )
```
2. Register a listener for `MemberUpdateEvent`

3. Run the bot, invite it to your server, and assign yourself a role (thus triggering a `MemberUpdateEvent`)

4. Using a debugger tool, inspect the contents of the `MemberUpdateEvent` and see that its `private final Member old;` element is null.

**Expected Behavior:** 
I expect a non-null value for `private final Member old;`. This is the object that contains the user's previous set of role IDs, which I can compare against the new set of role IDs to detect when someone has been given a special role.

**Actual Behavior:** 
`private final Member old;` is null, which prevents me from being able to determine what triggered the `MemberUpdateEvent`.

**Version:** 
3.1.1

**Other:** 

![](https://cdn.discordapp.com/attachments/451125724766535710/776194461251928095/unknown.png)

**Comment by quanticc** [comment]
I was able to reproduce with offline members

**Comment by quanticc** [comment]
Scratch that, any guild without member requesting will be affected (all non-large ones by default)

Considering if we should just warn+suggest explicit requests, or change the behavior, although this can have a negative effect on the ws outbound rate limit.

**Comment by quanticc** [comment]
Workaround for now: use MemberRequestFilter to all, or a different one you can customize.

```java
DiscordClient.create(System.getenv("token"))
        .gateway()
        .setMemberRequestFilter(MemberRequestFilter.all())
        .setEnabledIntents(IntentSet.of(
                Intent.GUILD_MEMBERS,
                Intent.GUILDS,
                Intent.GUILD_MESSAGES,
                Intent.GUILD_MESSAGE_REACTIONS,
                Intent.DIRECT_MESSAGES,
                Intent.DIRECT_MESSAGE_REACTIONS))
        .withGateway(gateway -> gateway.on(MemberUpdateEvent.class,
                event -> Mono.fromRunnable(() -> {
                    Member old = event.getOld().orElse(null);
                    System.out.println("Old member: " + old);
                })))
        .block();
```

**Comment by j0rdanit0** [comment]
Confirmed that using `MemberRequestFilter.all()` works. So the drawback to this workaround is that this could cause rate limit issues?

**Comment by quanticc** [comment]
Yes, if your bot belongs to many small guilds, as 1 member list request is sent per guild. You could hit a rate limit and get successive members at a slower rate during this startup phase.

The alternative is to use `.none()` but you then depend on either D4J requesting them on demand, or having to proactively call `requestMembers` variants

**Comment by j0rdanit0** [comment]
Yeah, unfortunately, this workaround causes startup issues on my prod bot, as you predicted. I'll need to wait for a more permanent solution. Thanks!
